llvm/include/llvm/IR/Intrinsics.td

```diff
@@ -2362,13 +2362,32 @@ def int_load_relative: DefaultAttrsIntrinsic<[llvm_ptr_ty], [llvm_ptr_ty, llvm_a
 def int_asan_check_memaccess :
   Intrinsic<[],[llvm_ptr_ty, llvm_i32_ty], [ImmArg<ArgIndex<1>>]>;
 
+// HWASan intrinsics to test whether a pointer is addressable.
+//===----------------------------------------------------------------------===//
+// Parameters: Shadow base (passed in a register), pointer to be checked for validity, AccessInfo
```

Comment:
Move this below Variant 1? (And copy to Variant 2?, And add Parameters to Variant 4?)

Reply:
Done